### PR TITLE
Laboratory work #4, Shkunov Pavel - 21FPL2 

### PR DESCRIPTION
**Pull request by pa-shk**

### CODE REVIEW

**Review by artyomtugaryov — APPROVED** [review]

File: lab_4_summarization_textrank/main.py [hunk 1]
```diff
@@ -84,59 +119,90 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
         """
-        pass
+        check_type(text, [str])
+        text = text.replace('\n', ' ').replace('  ', ' ')
+        sentences = []
+        count = 0
+        start = 0
+        punctuation_index = 0
+        space_flag = False
+        for index, elem in enumerate(text):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:141]: .

File: lab_4_summarization_textrank/main.py [hunk 2]
```diff
@@ -84,59 +119,90 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
     def _learn_indices(self, tokens: tuple[str, ...]) -> None:
         """
         Fills attributes mapping words and integer equivalents to each other
         :param tokens: a sequence of string tokens
         :return:
         """
-        pass
+        check_iterable(tokens, [tuple], [str])
+        new_tokens = (i for i in tokens if i not in self._word2id)
+        for index, token in enumerate(new_tokens, start=max(1000, 1000 + len(self._word2id))):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:192]: magic

File: lab_4_summarization_textrank/main.py [hunk 3]
```diff
@@ -184,7 +257,17 @@ def add_edge(self, vertex1: Sentence, vertex2: Sentence) -> None:
         :param vertex2:
         :return:
         """
-        pass
+        if vertex1 == vertex2:
+            raise ValueError
+        for vertex in vertex1, vertex2:
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:262]: too nested

File: lab_4_summarization_textrank/start.py [hunk 4]
```diff
@@ -27,6 +31,27 @@
...
+    print('Encoded:', *(sentence.get_encoded() for sentence in SENTENCES))
+    MATRIX = SimilarityMatrix()
+    MATRIX.fill_from_sentences(SENTENCES)
+    RANKING = TextRankSummarizer(MATRIX)
+    RANKING.train()
+    SUMMARY = RANKING.make_summary(5)
+    print('Summary:\n', SUMMARY)
+
+    BUDDY = Buddy(paths_to_texts, stop_words, punctuation, idf)
+    QUIRES = ['Когда родился Гагарин?', 'Кто такой Ленин?', 'Где работает Демидовский?']
```
Comment (demid5111) [inline lab_4_summarization_textrank/start.py:48]: query

**Pull request closed without merging**